### PR TITLE
Use manifest.zip with 3.8 file if found

### PR DESCRIPTION
**Pull request by cigamit**
##### SUMMARY
Adds in code to look for a manifest.zip file, if found it will use it and AT 3.8, else it falls back to using the tower_license and AT 3.7

We should add code in to look for a tower_manifest variable also, incase we need to pass it in (base64 encoded ofcourse) that way.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
- provisioner

### CODE REVIEW

**Comment by cigamit** [comment]
If we decide to just drop tower_license support in the future and go with manifests, this becomes much simpler.

File: provisioner/roles/control_node/tasks/tower.yml [hunk 1]
```diff
@@ -114,3 +114,23 @@
...
+      src: "{{ playbook_dir }}/manifest.zip"
+    register: manifest_file
+
+  - name: Post manifest file
+    uri:
+      url: https://{{ansible_host}}/api/v2/config/
+      method: POST
+      user: admin
+      password: "{{admin_password}}"
+      body: '{ "eula_accepted": true, "manifest": "{{ manifest_file.content }}" }'
```
Comment (liquidat) [inline provisioner/roles/control_node/tasks/tower.yml:132]: Out of interest, any special reason why you don't use a lookup plugin here to get rid of the previous step?

Comment (cigamit) [reply provisioner/roles/control_node/tasks/tower.yml:132]: Lookup plugins don't work with binary data.

**Comment by IPvSean** [comment]
this is good.... I need to rethink a bit of it... one of the problems is the `default_url` is the extra_var that the CI testing system is using.  So the way this seems to work, is that if we pass Tower 3.8 to the provisioner, it will check to see if you have a valid manifest, if you don't, it will revert it to 3.7... which is good but we might WANT it to fail b/c 3.8 doesn't work

also there is some lint problems here->
```
./provisioner/provision_lab.yml
  82:19     error    too many spaces after colon  (colons)
  83:24     error    too many spaces after colon  (colons)

./provisioner/roles/workshop_check_setup/defaults/main.yml
  1:1       warning  missing document start "---"  (document-start)
  1:20      error    no new line character at the end of file  (new-line-at-end-of-file)

./provisioner/roles/workshop_check_setup/tasks/main.yml
  101:28    error    too many spaces after colon  (colons)
  116:24    error    too many spaces after colon  (colons)

./provisioner/roles/control_node/tasks/tower.yml
  120:3     error    wrong indentation: expected 4 but found 2  (indentation)
```

I can see if I can fix it from my side

**Comment by IPvSean** [comment]
recheck

**Comment by IPvSean** [comment]
error is with capacity issues in AWS region
```
  msg: 'Instance creation failed => InsufficientInstanceCapacity: We currently do not have sufficient t2.2xlarge capacity in the Availability Zone you requested (us-east-1a). Our system will be working on provisioning additional capacity. You can currently get t2.2xlarge capacity by not specifying an Availability Zone in your request or choosing us-east-1b, us-east-1c, us-east-1d, us-east-1e, us-east-1f.'
```

going to merge because it is not a PR problem

**Comment by IPvSean** [comment]
thank you for supporting the github.com/ansible/workshops project @cigamit 

thank you sir

**Pull request merged**